wagtail/contrib/forms/tests/test_views.py

```diff
@@ -1289,6 +1289,17 @@ def construct_forms_for_user(user, queryset):
             reverse("wagtailforms:list_submissions", args=(self.form_page.id,)),
         )
 
+    def test_delete_submission_with_next_url(self):
+        next_url = "/"
```

Comment:
To cover https://github.com/wagtail/wagtail/issues/12007#issuecomment-2165095879, we should add another test to use the form submissions index URL with the correct `p` parameter as the `next` parameter, and remove the correct submissions to make it so that the page would be out of range. Then, we assert that the redirects eventually result in a 200 response instead of 404, which I think `assertRedirects` already does.

And as per that comment, we should revert 8e106f40c6acb9abf0a94238c8094950d0756b68 to make it work.